BBC.js

```diff
@@ -46,20 +46,37 @@ function detectWeb(doc, url) {
 		}
 		return "newspaperArticle";
 	}
-	if(url.indexOf("/newsbeat/article") != -1)
+	if(url.indexOf("/newsbeat/article") != -1){
 		return "blogPost";
+	}
 	if (getSearchResults(doc, true)) {
 		return "multiple";
 	}
 }
 
 function getSearchResults(doc, checkOnly) {
+	var namespace = doc.documentElement.namespaceURI;
+	var nsResolver = namespace ? function(prefix) {
+	if (prefix == 'x') return namespace; else return null;
+	} : null;
+
 	var items = {};
 	var found = false;
 	var rows = ZU.xpath(doc, '//a[h3[@class="title-link__title"]]');
-	for (var i=0; i<rows.length; i++) {
+	//for NewsBeat
```

Comment:
do you have an example URL or could create a test here? I'm not quite happy with the code yet (would like to avoid the `ZU.xpath` together with the `doc.evaluate`) but I can't see where this applies exactly.